### DISCUSSION TITLE
Restrict access to all routes except for specific ones for a specific role

### DISCUSSION THREAD

**Discussion by AlexeyOnsite** [symfony/symfony#46093] [Q&A] [answered]
Hello,
I need to define a list of routes which are accesible by a user with a specific role. All ather routes need to be prohibited.
Is there a way to achieve it with symfony standard security config?

**Comment by alessandro-podo** [comment]
access_control:
        - { path: ^/login, roles: PUBLIC_ACCESS }
        - { path: ^/datenschutz, roles: ROLE_... }
        - ....
        - { path: ^/*, roles: ROLE_NO_ACCESS }
        
       something like this should work, do i missunderstand your question?

**Reply by AlexeyOnsite** [reply]
This could be surely possible, but then I have to define it for every route in my application and there are a lot. 
I've already achieved it with completely custom firewall configuration for symfony 3. It was quite cumbersome, like this:

    security.voter.restricted:
        class: FDB\Security\Voter\RestrictedAccessVoter

    security.access_decision_manager.restricted:
        class: Symfony\Component\Security\Core\Authorization\AccessDecisionManager
        arguments:
            - ["@security.voter.restricted"]

    security.inverse_request_matcher:
        class: FDB\Security\WhitelistInverseRequestMatcher
        arguments:
            - "%whitelist.restricted_access%"

    security.firewall.restricted.accessmap:
        class: Symfony\Component\Security\Http\AccessMap
        calls:
            - method: add
              arguments:
                  - "@security.inverse_request_matcher"
                  - ["ROLE_RESTRICTED_ACCESS"]

    security.firewall.listener.restricted:
        class: Symfony\Component\Security\Http\Firewall\AccessListener
        arguments:
            - "@security.token_storage"
            - "@security.access_decision_manager.restricted"
            - "@security.firewall.restricted.accessmap"
            - "@security.authentication_manager"

Now I'm porting the application to symfony 6 and would like to stick to a standard config.

**Reply by alessandro-podo** [reply]
if think you use the glob syntax for regex

**Reply by AlexeyOnsite** [reply]
I've managed to implement it with a combination of access_control and a custom voter. 
Thank you anyway for taking time!

**Comment by AlexeyOnsite** [comment, marked as answer]
This combination did the trick:

    access_decision_manager:
        strategy: unanimous
    ...
    access_control:
        - { path: ^/login, roles: PUBLIC_ACCESS }
        - { path: ^/register, roles: PUBLIC_ACCESS }
        - { path: ^/, roles: IS_AUTHENTICATED_FULLY }

And the custom voter which checks for url in a list and a certain role.
